quodlibet/util/collection.py

```diff
@@ -47,19 +48,73 @@
 ELPOEP = list(reversed(PEOPLE))
 PEOPLE_SCORE = [100 ** i for i in range(len(PEOPLE))]
 
+SMOOTHING_FACTOR_DEFAULT = 10
 
-def avg(nums):
+
+def unweighted_average(nums):
```

Comment:
Would you mind adding typing annotations to all new (and ideally touched) code? It helps in the long term

Reply:
Happy to add it!

I have to say though, I'm unsure whether the readability decrease is it worth it in simple cases:

```python
def weighted_average(nums: List[float], weights: List[float]) -> List[float]:
```

vs

```python
def weighted_average(nums, weights):
```

In general though, I do like `typing` :+1:

Reply:
I agree around the readability a bit, though have got used to it a bit more (and now use >79 chars pretty much everywhere which helps).

It's the ability to find bugs, and to document highly dynamic or confusing code for future people that I especially like here.

Not also those two inputs can be relaxed to `Iterable` here I think (and that's probably just `Number`, not necessarily float right?)

Reply:
> Number

Basically, yes. Have chosen `Real` though, which is just `Union[int, float]`, not the real [`Real`](https://docs.python.org/3/library/numbers.html#numbers.Real), as mypy has an open issue about these not working: https://github.com/python/mypy/issues/3186 (still tried but didn't work)

Reply:
I don't know much about your use case (of if you even consider this an issue worth pursuing further), but you might want to see if [``numerary``](/posita/numerary) can help you. If [this journey](https://posita.github.io/numerary/latest/whytho/) lines up with your own, ``numerary`` could be an intermediary solution until python/mypy#3186 is fixed. (Alternatively, the techniques used therein might provide inspiration if you can't take a dependency.) Happy to [consult here](https://github.com/posita/numerary/discussions), if helpful. Apologies for the interruption if this is a distraction for you.